### PR TITLE
New features

### LINKED ISSUE #219: bee run from different folder
I want to run a bee project from my home folder without using cd to enter folder but 'bee run projectName' doesnt work, I can't enter path, what should I do (I don't know if it is a issue)

### PR DESCRIPTION
**Pull request by amrfaissal**
* Added a Makefile
* Abbility to do `bee run` from either the current directory or outside the `$GOPATH` (Fixes #219)

### CODE REVIEW

**Comment by astaxie** [comment]
@amrfaissal please solve the conflict.

**Comment by amrfaissal** [comment]
@astaxie The conflict is fixed.

File: run.go [hunk 1]
```diff
@@ -53,49 +59,56 @@ func init() {
...
 		ColorLog("[INFO] Uses '%s' as 'appname'\n", appname)
 	} else {
-		appname = args[0]
+		gopathsrc := path.Join(gopath, "src")
+		currpath = path.Join(gopathsrc, args[0])
+		appname = path.Base(currpath)
+
+		// Check if passed Bee application path/name exists
+		// in $GOPATH/src workspace
+		if !isExist(currpath) {
```
Comment (astaxie) [inline run.go]: do we need to check all gopath? As gopath may include multi paths, I think we need to check all these paths

Comment (amrfaissal) [inline run.go]: @astaxie I think by convention and in case of multiple paths, it should default to the first directory in the list.

Comment (astaxie) [inline run.go]: No, convention the fisrt gopath always the `go get` path. Like me the first gopath is the vendor package path. the second path is my project gopath

Comment (amrfaissal) [inline run.go]: @astaxie I've added the check for multiple paths.

File: run.go [hunk 2]
```diff
@@ -53,51 +61,56 @@ func init() {
...
-	exit := make(chan bool)
-	crupath, _ := os.Getwd()
-
 	if len(args) == 0 || args[0] == "watchall" {
-		appname = path.Base(crupath)
+		currpath, _ = os.Getwd()
+
+		if !isBeegoProject(currpath) {
+			exitPrint(fmt.Sprintf("Bee does not support non Beego project: %s", currpath))
+		}
```
Comment (astaxie) [inline run.go]: inside SearchGOPATHs already has isBeegoProject. I think you can judge by the `currentGoPath`, if it's empty it should be not Beego project

Comment (amrfaissal) [inline run.go]: @astaxie Good catch! In this case isBeegoProject() is redundant. I've removed it.

**Pull request merged**